snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -919,16 +985,37 @@ def _organize_filesets(fileset, base_dir):
                 shutil.move(src, dst)
 
 
+def _sort_snap_dirs(snap_dirs):
+    new_snap_dirs = set()
+    for snap_dir in snap_dirs:
+        if type(snap_dir) == tuple:
+            snap_dir = snap_dir[1]
+        new_snap_dirs.add(snap_dir)
+
+    return sorted(new_snap_dirs, reverse=True)
+
+
 def _clean_migrated_files(snap_files, snap_dirs, directory):
     for snap_file in snap_files:
-        os.remove(os.path.join(directory, snap_file))
+        path = snap_file
+        if type(snap_file) == tuple:
+            path = snap_file[1]
+
+        # XXX: hack to remove directories should be fixed elsewhere
+        if os.path.isdir(os.path.join(directory, path)):
```

Comment:
That's exactly my concern though. Doesn't that mean that below you'll try to call `os.rmdir()` on a symlink? What removes the symlink itself?